src/Stratis.Bitcoin.Features.PoA/Voting/ReconstructFederationService.cs

```diff
@@ -33,6 +35,12 @@ public void Reconstruct(int height)
                 return;
             }
 
+            if (this.isBusyReconstructing)
+            {
+                this.logger.Info($"Reconstructing the federation is already underway.");
+                return;
+            }
+
```

Comment:
More than one thread can get past this `if` as the variable is only being set later. Testing and setting should ideally be enclosed locally within a `lock` here.